File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        Map<Integer, Integer> ggulMap = new HashMap<>();
        
        for(int tang: tangerine){
            ggulMap.put(tang, ggulMap.getOrDefault(tang, 0)+1);
        }
        
        List<Map.Entry<Integer, Integer>> list = new ArrayList<>(ggulMap.entrySet());
        list.sort((a, b) -> b.getValue().compareTo(a.getValue()));
        
        for(Map.Entry<Integer, Integer> map: list){
            answer += 1;
            k -= map.getValue();
            
            if(k <= 0){
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import defaultdict

def solution(k, tangerine):
    answer = 0
    
    ggul_dict = defaultdict(int)
    
    for tang in tangerine:
        ggul_dict[tang] += 1
    
    ggul_dict = dict(sorted(ggul_dict.items(), key=lambda x: -x[1]))
    
    for key, value in ggul_dict.items():
        answer += 1
        k -= value
            
        if k <= 0:
            break      
    
    return answer